src/app/post-ad-component/post-ad-component.component.ts

```diff
@@ -120,10 +197,16 @@ export class PostAdComponentComponent implements OnInit {
       let image =new ImageProperty();
       this.imageArray.push(image);
     }
+    // If the file enterd was invalid delete the extra image holder created
+    if(err)
+      this.removeImage(id);
   }
  
   removeImage(id)
   {
+    //send the DELETE request and then remove from local
+    this.uploadImageService.deleteImage(this.productModel.imageUrls[id]).subscribe();
```

Comment:
`subscribe`? do we need it when no action is to be taken

Reply:
Yes, it is needed to send the delete request to the server.